### PR TITLE
Implement additional ONS_CIS columns

### PR DESCRIPTION
**Pull request by rebkwok**
See #972 

- Use ONS_CIS_New table instead of ONS_CIS (ONS_CIS_New should contain the same data as ONS_CIS, but with some additional columns)
- Add the columns that are needed for the ONS mental health study:
    - visit_date
    - visit_num
    - nhs_data_share
    - last_linkage_dt

`nhs_data_share` is an int column in the table, but ONS have confirmed that it can take one of two values, 0 ("opt-in") or 1("opt-out").  The `ons_cis` table is a QueryTable that converts the `nhs_data_share` column to its corresponding label.  This avoids issues with 0 values which actually represent a data value, and makes the output similar to cohort-extractor.  The ONS mental health study will combine ons cis outputs from cohort-extractor and databuilder, so we'll want them to be consistent in the way they output.

### CODE REVIEW

File: databuilder/tables/beta/tpp.py [hunk 1]
```diff
@@ -195,5 +195,5 @@ class ons_cis(EventFrame):
 
     visit_date = Series(datetime.date)
     visit_num = Series(int)
-    nhs_data_share = Series(int)
+    nhs_data_share = Series(str)
```
Comment (evansd) [inline databuilder/tables/beta/tpp.py:198]: This should have a categorical constraint if we're going to model it as a categorical. Something like:
```py
nhs_data_share = Series(
    str,
    constraints=[
        Constraint.Categorical(["opt-out", "opt-in"]),
    ],
)
```

But I wonder if this wouldn't be better modelled as a boolean. Something like `is_opted_in_to_nhs_data_share`?

**Review by evansd — APPROVED** [review]

File: databuilder/tables/beta/tpp.py (hunk 1 above)
Comment (rebkwok) [reply databuilder/tables/beta/tpp.py:198]: Yes, a boolean probably makes sense.  I wonder if that would make it confusing for the ONS team who are using a mix of cohort-extractor and databuilder?
I guess we can ask them

Comment (evansd) [reply databuilder/tables/beta/tpp.py:198]: Well that's neat! (I think you meant "0=opt-in" above, right?)

Comment (rebkwok) [reply databuilder/tables/beta/tpp.py:198]: yes :facepalm:

Comment (rebkwok) [reply databuilder/tables/beta/tpp.py:198]: I will check the db to make sure there really are only 0, 1 or null in that column first though

Comment (rebkwok) [reply databuilder/tables/beta/tpp.py:198]: Confirmed :point_up:

**Review by evansd — APPROVED** [review]

**Pull request merged**